### PR TITLE
Code Insights: [FE] Connect capture group creation page to api

### PR DESCRIPTION
**Pull request by vovakulikov**
Closes https://github.com/sourcegraph/sourcegraph/issues/28026

This PR is the first PR of three for finishing capture group creation UI
- This one
- https://github.com/sourcegraph/sourcegraph/pull/28855
- https://github.com/sourcegraph/sourcegraph/pull/28861

This PR does a few things 
- Updates GQL API (supports capture group insight for creating/updating mutations)
- Adds capture group insight interface 
- Connects the capture group insight page to GQL API

## How to test
- Go to /insights/create/capture-group
- Fill in the form and click create insight
- See in the console that network request with create mutation has appeared in the network tab
- Check your "All insights dashboard"

### CODE REVIEW

File: client/web/src/enterprise/insights/components/creation-ui-kit/sanitizers/repositories.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+ * Returns parsed by string repositories list.
+ *
+ * @param rawRepositories - string with repositories split by commas
+ */
+export function getSanitizedRepositories(rawRepositories: string): string[] {
+    return rawRepositories
+        .trim()
+        .split(/\s*,\s*/)
+        .filter(repo => repo)
+}
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/components/creation-ui-kit/sanitizers/repositories.ts:11]: Definitely don't need this now, but I would love to see a unit test for this. Something that really demonstrates what we are guarding against.

File: client/web/src/enterprise/insights/core/backend/gql-api/methods/create-insight.ts [hunk 2]
```diff
@@ -66,15 +69,16 @@ export const createInsight = (apolloClient: ApolloClient<object>, input: Insight
...
-    const input: LineChartSearchInsightInput = prepareSearchInsightCreateInput(insight, dashboard)
+    const input = getInsightGqlInput(insight, dashboard)
 
-    // In case if we want to create insight with some predefined filters we have to
-    // crate insight first and only then update this insight with filters value
-    // This is lack of our API flexibility and should be fixed as soon as BE gql API
-    // support filters in create insight mutation.
+    // In case if we want to create an insight with some predefined filters we have to
+    // create the insight first and only after update this newly created insight with filter values
+    // This is doe to lack of gql API flexibility and should be fixed as soon as BE gql API
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/core/backend/gql-api/methods/create-insight.ts:79]:
```suggestion
    // This is due to lack of gql API flexibility and should be fixed as soon as BE gql API
```

**Review by unclejustin — APPROVED** [review]
@vovakulikov LGTM 🌮  just one typo and a small observation about unit testing later.

File: client/web/src/enterprise/insights/components/creation-ui-kit/sanitizers/repositories.ts (hunk 1 above)
Comment (vovakulikov) [reply client/web/src/enterprise/insights/components/creation-ui-kit/sanitizers/repositories.ts:11]: Thanks for thinking about it. This is the problem that I usually forget about small functions unit tests. I added a few test cases for it.

**Pull request merged**